core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -148,6 +152,59 @@ private[spark] class EventLoggingListener(
     }
   }
 
+  /** cut the event log file **/
+  private def logReload(): Unit = {
+    val tlogPath = getLogPath(logBaseDir, appId, appAttemptId, compressionCodecName, cutType)
+    // ����ȡ��logPathû�䣬�Ͳ������������־�ļ�
+    if (tlogPath == logPath) {
+      return
+    }
+
+    if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDirectory) {
+      throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
+    }
+
+    val workingPath = tlogPath + IN_PROGRESS
+    val path = new Path(workingPath)
+    val uri = path.toUri
+    val defaultFs = FileSystem.getDefaultUri(hadoopConf).getScheme
+    val isDefaultLocal = defaultFs == null || defaultFs == "file"
+
+    if (shouldOverwrite && fileSystem.delete(path, true)) {
+      logWarning(s"Event log $path already exists. Overwriting...")
+    }
+
+    /* The Hadoop LocalFileSystem (r1.0.4) has known issues with syncing (HADOOP-7844).
+     * Therefore, for local files, use FileOutputStream instead. */
+    val dstream =
+      if ((isDefaultLocal && uri.getScheme == null) || uri.getScheme == "file") {
+        new FileOutputStream(uri.getPath)
+      } else {
+        hadoopDataStream = Some(fileSystem.create(path))
+        hadoopDataStream.get
+      }
```

Comment:
From line 163 to line 185, this seems to be a blind copy of the body of `def start()`.
Please try to reduce this kind of copy&paste.